kyo-core/shared/src/main/scala/kyo/StreamCoreExtensions.scala

```diff
@@ -189,6 +195,333 @@ object StreamCoreExtensions:
             f: Frame
         ): Stream[V, Abort[E] & S & Async] =
             other.mergeHaltingLeft(stream)(using i1, i2, sct, t, f)
+
+        /** Applies effectful transformation of stream elements asynchronously, mapping them in parallel. Preserves chunk boundaries.
+          *
+          * @param parallel
+          *   Maximum number of elements to transform in parallel at a time
+          * @param bufferSize
+          *   Size of buffer used to mediate
+          * @param f
+          *   Asynchronous transformation of stream elements
+          */
+        def mapPar[V2, S2](parallel: Int, bufferSize: Int = defaultAsyncStreamBufferSize)(f: V => V2 < (Abort[E] & Async & S2))(
+            using
+            t1: Tag[Emit[Chunk[V]]],
+            t2: Tag[Emit[Chunk[V2]]],
+            t3: Tag[V2],
+            i1: Isolate.Contextual[S & S2, IO],
+            i2: Isolate.Stateful[S & S2, Abort[E] & Async],
+            ev: SafeClassTag[E | Closed],
+            frame: Frame
+        ): Stream[V2, Abort[E] & Async & S & S2] =
+            val initialState: (Fiber[E | Closed, Unit], Int) = (Fiber.unit, parallel)
+            Stream[V2, S & S2 & Abort[E] & Async]:
+                Channel.init[Maybe[Chunk[V2]]](bufferSize).map { channel =>
+                    AtomicInt.init(0).map: parAdjustmentRef =>
+                        val background = Async.run:
+                            val handledStream = ArrowEffect.handleState(t1, initialState, stream.emit)(
+                                handle = [C] =>
+                                    (input, state, cont) =>
+                                        val (prevEmitFiber, remainingEmitPar) = state
+
+                                        Loop(Fiber.success[E | Closed, Chunk[V2]](Chunk.empty[V2]), input, remainingEmitPar):
+                                            (prevChunkFiber, remainingChunk, remainingPar) =>
+                                                parAdjustmentRef.getAndSet(0).map: parAdjustment =>
+                                                    val adjustedRemainingPar = remainingPar + parAdjustment
+                                                    val nextParSection       = remainingChunk.take(adjustedRemainingPar)
+
+                                                    val nextChunkEffect = Async.foreach(nextParSection)(f)
+
+                                                    val newRemainingPar   = adjustedRemainingPar - nextParSection.size
+                                                    val newRemainingChunk = remainingChunk.drop(adjustedRemainingPar)
+
+                                                    if newRemainingPar <= 0 && newRemainingChunk.size <= 0 then
+                                                        nextChunkEffect.map: nextChunk =>
+                                                            prevChunkFiber.get.map: prevChunk =>
```

Comment:
```suggestion
                                                            prevChunkFiber.use: prevChunk =>
```
`use` is more lightweight. In performance-sensitive code, it's always best to use it, even if the value is discarded (`.use(_ => ...)`). There are other instances in the code.